### PR TITLE
zero-shot-classification page

### PR DESCRIPTION
**Pull request by SriPrarabdha**
- Added the placeholder for zero shot classification
- Added summary
Partially fixes https://github.com/huggingface/hub-docs/issues/361

### CODE REVIEW

File: tasks/src/zero-shot-classification/about.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+### Latent Embedding
+
+A word embedding is a learned representation for text where words that have the same meaning have a similar representation. individual words are represented as real-valued vectors in a predefined vector space. 
+
+So if we are given a sentence we can convert all it's words as embedding vector and take a mean of all the vectors to get a representation of sentence and given the labels like ['postive' , 'negative' , 'politics' , 'science' ] on which we want to classify and run them to the simailar text encoder like BERT
+
+By defining a similarity parameter we can see the similarity between embedding of our sentence and that of labels and return the label with most similarity
+
+In this way we can eaisly classify sentences without training the model
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:17]:
```suggestion
This way, we can easily classify sentences without training a separate model.

```
(maybe we can have this in previous paragraph)

File: tasks/src/zero-shot-classification/about.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+#### Inference
+
+We can use pretrained BERT model from
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:21]:
```suggestion
We can use pre-trained BERT model to create the embeddings, as follows:

```

File: tasks/src/zero-shot-classification/about.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+#### Inference
+
+We can use pretrained BERT model from 
+
+```python
+from transformers import pipeline
+
+encoder = pipeline("feature-extraction" , model = "deepset/sentence_bert")
+
+sentence = "Artificial Science will revolutnize the world"
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:28]:
```suggestion
sentence = "Artificial intelligence will revolutionize the world."

```

File: tasks/src/zero-shot-classification/about.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+In this way we can eaisly classify sentences without training the model
+
+#### Inference
+
+We can use pretrained BERT model from 
+
+```python
+from transformers import pipeline
+
+encoder = pipeline("feature-extraction" , model = "deepset/sentence_bert")
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:26]:
```suggestion
pipe = pipeline("feature-extraction", model = "deepset/sentence_bert")

```

File: tasks/src/zero-shot-classification/about.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+sentence = "Artificial Science will revolutnize the world"
+labels = ['postive' , 'negative' , 'politics' , 'science' ]
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:29]:
```suggestion
labels = ['positive' , 'negative' , 'politics' , 'science' ]

```
Maybe they're a bit irrelevant, only having news categories is less confusing.

File: tasks/src/zero-shot-classification/about.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+#Mean Pooling across sentence
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:31]:
```suggestion
# mean pooling across sentence

```
Can you explain what mean pooling is?

File: tasks/src/zero-shot-classification/about.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+sentence_vector = np.array(encoder(sentence).mean(axis=1))
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:32]:
```suggestion
sentence_vector = np.array(pipe(sentence).mean(axis=1))

```
Maybe explain this code a little, also import numpy above

File: tasks/src/zero-shot-classification/about.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+label_vector = [np.array(encoder(l).mean(axis=1)) for l in labels]
+
+from scipy import spatial
+#Using cosine similarity between sentence vector and label vector
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:36]:
```suggestion
# using cosine similarity to assess the similarity between the sentence vector and the label vector

```

File: tasks/src/zero-shot-classification/about.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+dist = [spatial.distance.cosine(sentence_vector , lv) for lv in label_vector]
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:37]:
```suggestion
similarity = [spatial.distance.cosine(sentence_vector , lv) for lv in label_vector]

```

File: tasks/src/zero-shot-classification/about.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+print(labels[np.argmin(dist)])
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:39]:
```suggestion
print(labels[np.argmin(similarity)])

```

File: tasks/src/zero-shot-classification/about.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+```
+## Natural Language Inference (NLI)
+
+In NLI the model determines the relationship between two given texts. Concretely, the model takes a premise and a hypothesis and returns a class that can either be:
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:44]:
```suggestion
In natural language inference, the model determines the relationship between two given texts. Concretely, the model takes a premise and a hypothesis and returns a class that can either be:

```

File: tasks/src/zero-shot-classification/about.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+- **entailment**, which means the hypothesis is true.
+- **contraction**, which means the hypothesis is false.
+- **neutral**, which means there's no relation between the hypothesis and the premise.
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:48]: Maybe explain what the hypothesis and premise is above 🙂

File: tasks/src/zero-shot-classification/about.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+Hugging face has implemented a Zero-Shot Classification pipeline which travels a slightly different route . It uses Natural Language Inference models to carry out classification by a bit of creative prompt engineering
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:50]:
```suggestion
The implementation of `zero-shot-classification` pipeline of Hugging Face is different. It uses Natural Language Inference models to carry out classification by a bit of creative prompt engineering.

```
Can you explain prompt engineering? Also different than what?

Comment (merveenoyan) [reply tasks/src/zero-shot-classification/about.md:50]: Maybe you could give link to sentence-similarity task page which is what you wrote the code for above.

File: tasks/src/zero-shot-classification/about.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+#### Inference
+
+```python
+from transformers import pipeline
+
+classifier = pipeline("zero-shot-classification" , model="facebook/bart-large-mnli")
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:57]:
```suggestion
pipe = pipeline("zero-shot-classification" , model="facebook/bart-large-mnli")

```

File: tasks/src/zero-shot-classification/about.md [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+sentence = "Artificial Science will revolutnize the world"
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:59]:
```suggestion
sentence = "Artificial intelligence will revolutionize the world."

```

File: tasks/src/zero-shot-classification/about.md [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+labels = ['postive' , 'negative' , 'politics' , 'science' ]
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:60]:
```suggestion
labels = ['positive' , 'negative' , 'politics' , 'science' ]

```
The same applies to this one.

File: tasks/src/zero-shot-classification/about.md [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+
+classifier(sentence , labels)
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:62]:
```suggestion
pipe(sentence , labels)

```

File: tasks/src/zero-shot-classification/about.md [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+
+## Useful Resources
+
+Would you like to learn more about the topic? Awesome! Here you can find some curated resources that you may find helpful!
+
+- [Github Repositery contaning all important papers , datasets and other resource in ZSL](https://github.com/sbharadwajj/awesome-zero-shot-learning)
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/about.md:70]:
```suggestion
- [A GitHub repository that contains important papers, datasets and more about zero-shot classification](https://github.com/sbharadwajj/awesome-zero-shot-learning)

```

File: tasks/src/zero-shot-classification/data.ts [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+						label: "NEGATIVE",
+						score: 0.008,
+					},
+				],
+			},
+		],
+	},
+	
+	
+	summary: "Zero-shot learning refers to a specific use case of deep learning where you want the model to classify data based on very few or even no labeled example",
```
Comment (merveenoyan) [inline tasks/src/zero-shot-classification/data.ts:44]:
```suggestion
	summary: "Zero-shot learning is the task of classifying text, based on few or no labeled examples.", 

```

**Review by merveenoyan — COMMENTED** [review]
Thanks a lot for working on this! Can you check my suggestions 🙂

File: tasks/src/zero-shot-classification/about.md [hunk 20]
```diff
@@ -0,0 +1,71 @@
+## Use Cases
+
+###  Human Level understanding of Text
+
+In pretrained language models like BERT , model is to finetuned according to the kind of task we want our model do to like sentimental analysis by providing a proper dataset for specific task . But us humans can tell wether a news article is postive or negative , related to science or politics without fine-tuning it . Similar to that a trained Zero Shot Model aims to perform any general task on a language
```
[the remark this answers is not in the document]
Comment (SriPrarabdha) [reply tasks/src/zero-shot-classification/about.md:5]: so should I remove BERT or explain it

[the remark this answers is not in the document]
Comment (SriPrarabdha) [reply tasks/src/zero-shot-classification/about.md:5]: Should I add a glossary?

File: tasks/src/zero-shot-classification/about.md [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+
+In pretrained language models like BERT , model is to finetuned according to the kind of task we want our model do to like sentimental analysis by providing a proper dataset for specific task . But us humans can tell wether a news article is postive or negative , related to science or politics without fine-tuning it . Similar to that a trained Zero Shot Model aims to perform any general task on a language
+
+## Task Variants 
+
+### Latent Embedding
+
+A word embedding is a learned representation for text where words that have the same meaning have a similar representation. individual words are represented as real-valued vectors in a predefined vector space. 
+
+So if we are given a sentence we can convert all it's words as embedding vector and take a mean of all the vectors to get a representation of sentence and given the labels like ['postive' , 'negative' , 'politics' , 'science' ] on which we want to classify and run them to the simailar text encoder like BERT
```
[the remark this answers is not in the document]
Comment (SriPrarabdha) [reply tasks/src/zero-shot-classification/about.md:17]: Thanks for the suggestions . I'll try to include all

**Comment by SriPrarabdha** [comment]
Thanks for helping me out with all the mistakes. It's my first open source contribution so this helps quite alot.

File: tasks/src/zero-shot-classification/about.md (hunk 6 above)
Comment (merveenoyan) [reply tasks/src/zero-shot-classification/about.md:31]: I see this comment is not addressed 🙂

**Review by merveenoyan — COMMENTED** [review]
From my previous review there's only couple of rephrasing left (can you take a look at the comments that I didn't resolve?) 🙂 and I'll be happy to merge

**Comment by merveenoyan** [comment]
I resolved comments that I think addressed, there's few comments left and I'll be happy to merge afterwards! Let me know if you need any help.

**Comment by merveenoyan** [comment]
@SriPrarabdha do you need any help?

**Comment by merveenoyan** [comment]
@SriPrarabdha can you solve merge conflicts and I will merge this PR 🙂🙂

**Comment by merveenoyan** [comment]
@SriPrarabdha do you need help with this one?

**Comment by merveenoyan** [comment]
Hey @SriPrarabdha 👋 
We have moved the codes for Tasks project to [huggingface.js repository](https://github.com/huggingface/huggingface.js) so if you feel like continuing this pull request you can move your work over there. Many thanks! 🤗
